lib/plugins/aws/provider/awsProvider.js

```diff
@@ -279,52 +286,62 @@ class AwsProvider {
     }
 
     const request = this.requestQueue.add(() =>
-      persistentRequest(() => {
-        if (options && !_.isUndefined(options.region)) {
-          credentials.region = options.region;
-        }
-        const awsService = new that.sdk[service](credentials);
-        const req = awsService[method](params);
-
-        // TODO: Add listeners, put Debug statments here...
-        // req.on('send', function (r) {console.log(r)});
+      persistentRequest(() =>
+        BbPromise.resolve()
+          .then(() => that.getCredentials())
```

Comment:
`BbPromise.resolve().then(() => that.getCredentials()).then(...` is unnecessarily verbose.

It can be just `that.getCredentials().then(...`